src/backtrace_support.rs

```diff
@@ -204,5 +204,57 @@ where
 /// Checks if a function is considered to be not in-app
 #[allow(unused)]
 pub fn is_sys_function(func: &str) -> bool {
-    WELL_KNOWN_SYS_MODULES.iter().any(|m| func.starts_with(m))
+    WELL_KNOWN_SYS_MODULES
+        .iter()
+        .any(|m| function_starts_with(func, m))
+}
+
+/// Checks whether the function name starts with the given pattern.
+///
+/// In trait implementations, the original type name is wrapped in "_< ... >" and colons are
+/// replaced with dots. This function accounts for differences while checking.
+pub fn function_starts_with(mut func_name: &str, pattern: &str) -> bool {
+    if func_name.starts_with("_<") {
+        func_name = &func_name[2..];
+    }
+
+    if !func_name.is_char_boundary(pattern.len()) {
+        return false;
+    }
+
+    func_name
```

Comment:
Chose a manual zero-copy compare implementation here, but it needs `chars` which might be slow. Maybe we should use `as_bytes()` and then compare (and accept false positives)